examples/examples/runtime_update_async.rs

```diff
@@ -0,0 +1,68 @@
+/*
+	Copyright 2023 Supercomputing Systems AG
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+		http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/
+
+//! Example that shows how to detect a runtime update and afterwards update the metadata.
+use substrate_api_client::{
+	ac_primitives::{AssetRuntimeConfig, Config},
+	api_client::UpdateRuntime,
+	rpc::JsonrpseeClient,
+	rpc_api::RuntimeUpdateDetector,
+	Api, SubscribeEvents,
+};
+use tokio::select;
+use tokio_util::sync::CancellationToken;
+
+type Hash = <AssetRuntimeConfig as Config>::Hash;
+
+#[cfg(feature = "sync-examples")]
+#[tokio::main]
+async fn main() {
+	println!("This example is for async use-cases. Please see runtime_update_sync.rs for the sync implementation.")
+}
+
+#[cfg(not(feature = "sync-examples"))]
+#[tokio::main]
+async fn main() {
+	env_logger::init();
+
+	// Initialize the api.
+	let client = JsonrpseeClient::with_default_url().unwrap();
+	let mut api = Api::<AssetRuntimeConfig, _>::new(client).await.unwrap();
+
+	let subscription = api.subscribe_events().await.unwrap();
+	let mut update_detector: RuntimeUpdateDetector<Hash, JsonrpseeClient> =
+		RuntimeUpdateDetector::new(subscription);
+	println!("Current spec_version: {}", api.spec_version());
+	let detector_future = update_detector.detect_runtime_update();
```

Comment:
If I understood correctly, this example now detects exactly one runtime upgrade. I'm wondering if that's really a use case, or if we should rather provide a more "real world" alike one. 
E.g. actually running a while loop that detects and updates accordingly. What do you think?

Reply:
Yes I it detects one. I think I would leave it at that given that we now also trigger an actual update. I hope the user can implement a loop 😄